torch/distributions/gamma.py

```diff
@@ -56,8 +56,9 @@ def __init__(self, alpha, beta):
         elif alpha_num and beta_num:
             alpha, beta = torch.Tensor([alpha]), torch.Tensor([beta])
         elif alpha.size() != beta.size():
-            raise ValueError('Expected alpha.size() == beta.size(), actual {} vs {}'.format(
-                alpha.size(), beta.size()))
+            param_shape = broadcast_shape(alpha.size(), beta.size())
```

Comment:
The part with upcasting alpha and beta to tensors should be part of a helper as well (it is repeated in one of the other PRs)

Reply:
I was thinking of addressing that separately, but it is closely related to this; so let me fix the way we are handling scalars too (that should fix the tests).